src/transformers/models/t5/tokenization_t5.py

```diff
@@ -301,14 +301,15 @@ def __setstate__(self, d):
         self.sp_model.Load(self.vocab_file)
 
     def tokenize(self, text: "TextInput", **kwargs) -> List[str]:
-        if not text.startswith(" "):
-            text = " " + text
+        text = SPIECE_UNDERLINE + text
         return super().tokenize(text, **kwargs)
```

Comment:
we always prepend with `SPIECE_UNDERLINE`. That way we can know id we have to post process the string or not. 
This is in order to `hack` the `_tokenize` method, which does not know if we are at the beginning of the string or if we are after a special token, which eats anything